ScienceJournal/ToneGenerator/ScaleSoundType.swift

```diff
@@ -28,8 +28,13 @@ class ScaleSoundType: PitchedSoundType {
                           valueMin: Double,
                           valueMax: Double,
                           timestamp: Int64) -> Double? {
-    let index = Int(floor((value - valueMin) / (valueMax - valueMin) * Double(pitches.count - 1)))
-    return frequency(from: Double(pitches[index]))
+    let floorValue = floor((value - valueMin) / (valueMax - valueMin) * Double(pitches.count - 1))
+    let index = Int(exactly: floorValue)
+    if let index = index {
```

Comment:
I think in this case the extra index var is more confusing than it is in other instances.
```suggestion
    if let index = Int(exactly: floorValue) {
```

Reply:
Good point, updating